lib/std/meta/trait.zig

```diff
@@ -145,6 +145,49 @@ test "isSliceOf" {
     try testing.expect(!isSliceOf(.Struct)([][]struct {}));
 }
 
+/// Returns true if the function takes its enclosing container type as the first argument
+pub fn isMethod(comptime name: []const u8) TraitFn {
+    const Closure = struct {
+        pub fn trait (comptime T: type) bool {
+            if (!comptime isContainer(T)) return false;
+            if (!comptime @hasDecl(T, name)) return false;
+            const DeclType = @TypeOf(@field(T, name));
+            switch (@typeInfo(DeclType)) {
+                .Fn => |f| {
+                    if (f.params.len == 0) return false;
+                    const Type = f.params[0].type orelse return false;
+                    switch (@typeInfo(Type)) {
+                        .Pointer => |ptr| {
+                            return ptr.child == T and ptr.size == .One;
+                        },
+                        else => return Type == T,
```

Comment:
Aside from just `T` and `*T`/`*const T` as the receiver type, there are other possibilities you may want to consider here: https://github.com/ziglang/zig/blob/1880c799aebed7efe524c040998b27a2b361ff43/test/behavior/fn.zig#L445-L479 Specifically, `?T`, `?*T`/`?*const T`, and `E!T` (where `E` is any error set). Additionally `[*c]T`/`[*c]const T` are supported, but not part of the linked behavior test: https://github.com/ziglang/zig/blob/1880c799aebed7efe524c040998b27a2b361ff43/src/Sema.zig#L26565-L26566

Reply:
That's fair. I can add those.